lib/lcevc/lcevc_dil.js

```diff
@@ -0,0 +1,277 @@
+      this.media_.style.display = 'none';
+
+      this.parent_ = this.media_.parentElement;
+      if (this.parent_) {
+        this.lcevcContainer_.appendChild(this.canvas_);
+        this.parent_.appendChild(this.lcevcContainer_);
+        if ('ResizeObserver' in window) {
+          this.resizeObserver_ = new ResizeObserver((_entries) => {
+            window.requestAnimationFrame(() => {
+              this.playerResize_();
+            });
+          });
+          this.resizeObserver_.observe(this.parent_);
+        }
+        this.playerResize_();
+      }
+    }
+  }
+
+  /**
+   * Resizing the container based on the source video dimensions.
+   * @private
+   */
+  playerResize_() {
+    const lcevcContainerParent = /** @type {HTMLElement} */
+      (this.lcevcContainer_.parentElement);
+
+    if (this.isSupported_()) {
+      // We try to find the actual aspect ratio with the lcevc enhanced frame
+      // if this value is not a valid value we default it to 1920x1080
+      const aspectRatio = (this.dil_ && this.dil_.aspectRatio > 0 ?
+        this.dil_.aspectRatio :
+        this.media_.videoWidth / this.media_.videoHeight) || 1920 / 1080;
+      this.lcevcContainer_.style.width =
+        `${(lcevcContainerParent).offsetWidth}px`;
+      if (this.dil_ && this.dil_.isFullscreen) {
+        this.lcevcContainer_.style.height = `${window.innerHeight}px`;
+        if (this.canvas_.height < this.lcevcContainer_.offsetHeight) {
+          this.canvas_.style.top =
+            `${(window.innerHeight - this.canvas_.height) / 2}px`;
+        } else {
+          this.canvas_.style.top = '';
+          this.canvas_.style.height = `${window.innerHeight}px`;
+        }
+      } else {
+        this.lcevcContainer_.style.height =
+          `${lcevcContainerParent.offsetWidth / aspectRatio}px`;
+
+        this.canvas_.style.height = `${this.lcevcContainer_.offsetHeight}px`;
+        this.canvas_.style.top = '';
+      }
+    }
+  }
+
+  /**
+   * Updates LCEVC DIL with the index of the variant that is being downloaded.
+   * @param {number} level // Index of currently active variant.
+   * @param {number} containerFormat // Container Format of the stream.
+   * @private
+   */
+  updateLevel_(level, containerFormat) {
+    if (this.dil_) {
+      this.level_ = level;
+      this.containerFormat_ = containerFormat;
+      this.dil_.setLevelSwitching(level, 1);
+      this.dil_.setContainerFormat(containerFormat);
+    }
+  }
+
+  /**
+   * Update current active variant
+   * @param {Array<shaka.extern.Track>} tracks
+   */
+  updateVariant(tracks) {
+    let containerFormat = shaka.lcevc.Dil.ContainerFormat.MPEG2_TS;
+    for (let iterator = 0; iterator < tracks.length; iterator += 1) {
+      if (tracks[iterator].active) {
+        switch (tracks[iterator].mimeType) {
+          case 'video/webm': {
+            containerFormat = shaka.lcevc.Dil.ContainerFormat.WEBM;
+            break;
+          }
+          case 'video/mp4': {
```

Comment:
No case for TS?  I see you have an enum defined for it.

Reply:
TS is the default value, removed from the enum